streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -100,37 +101,85 @@
     KStream<K, V> through(String topic);
 
     /**
+     * Materialize this stream to a topic, also creates a new instance of {@link KStream} from the topic
+     * using default serializers and deserializers.
+     * This is equivalent to calling {@link #to(String, StreamPartitioner)} and {@link org.apache.kafka.streams.kstream.KStreamBuilder#stream(String...)}.
+     *
+     * @param topic        the topic name
+     * @param partitioner  the function that should be used to determine the partition for each message processed by the sink
```

Comment:
Nit: users with the high-level DSL may not be familiar with "sink", so we should reword it from the Processor API. How about "the function used to determine how records are distributed among partitions of the topic"